src/mpi/pt2pt/recv.c

```diff
@@ -145,6 +145,25 @@ int MPI_Recv(void *buf, int count, MPI_Datatype datatype, int source, int tag,
         goto fn_exit;
     }
 
+    if (unlikely(MPIR_CVAR_ENABLE_FT &&
+                 !MPIR_Request_is_complete(request_ptr) &&
+                 MPID_Request_is_anysource(request_ptr) &&
+                 !MPID_Comm_AS_enabled(request_ptr->comm))) {
+        /* --BEGIN ERROR HANDLING-- */
+        MPID_Cancel_recv(request_ptr);
+        MPIR_STATUS_SET_CANCEL_BIT(request_ptr->status, FALSE);
+        MPIR_ERR_SET(request_ptr->status.MPI_ERROR, MPIX_ERR_PROC_FAILED, "**proc_failed");
+        mpi_errno = request_ptr->status.MPI_ERROR;
+        goto fn_fail;
+        /* --END ERROR HANDLING-- */
+    }
```

Comment:
Moving this block should be in a separate prereq patch.

Reply:
done.